manticore/ethereum/solidity.py

```diff
@@ -1,10 +1,36 @@
+
+from typing import Any, Dict, Mapping, Optional, Sequence, Iterable, Tuple
 import pyevmasm as EVMAsm
 
 from .abi import ABI
 from ..exceptions import EthereumError
+from ..utils.deprecated import deprecated
 
 
 class SolidityMetadata(object):
+
+    @staticmethod
+    def function_signature_for_name_and_inputs(name: str, inputs: Sequence[Mapping[str, Any]]) -> str:
+        """Returns the function signature for the specified name and Solidity JSON metadata inputs array.
+
+        The ABI specification defines the function signature as the function name followed by the parenthesised list of
+        parameter types separated by single commas and no spaces.
+        See https://solidity.readthedocs.io/en/latest/abi-spec.html#function-selector
+        """
+        return name + SolidityMetadata.tuple_signature_for_components(inputs)
+
+    @staticmethod
+    def tuple_signature_for_components(components: Sequence[Mapping[str, Any]]) -> str:
+        """Equivalent to ``function_signature_for_name_and_inputs('', components)``."""
+        ts = []
+        for c in components:
+            t: str = c['type']
+            if t.startswith('tuple'):
+                assert len(t) == 5 or t[5] == '['
```

Comment:
Shouldn't it be `len(t) >= 5`? as we use `t[5:]` later on

Reply:
No, since in this branch `len(t)` is always ≥ 5 because `t` starts with 'tuple' and the right side of the `or` expression should only be evaluated if the `t` has more than 5 characters.

Reply:
Oh it is `or` here. That makes sense :).